### DISCUSSION TITLE
Free relay and spam immediately after launch in docker

### DISCUSSION THREAD

**Discussion by perec007** [Mailu/Mailu#3072] [User support] [answered]
After I picked up and configured the mailer via Docker Compose, I discovered that it turns out that anyone on the Internet can simply send letters through port 25.

I checked and this is indeed true.
For example, like this, without authorization from a third-party server, I can send spam through my server
```
$ curl smtp://myserver.com --mail-from myself@myserver.com --mail-rcpt testmail@gmail.com --upload-file email.txt
$ cat email.txt
From: John Smith <myself@myserver.com>
To: Joe Smith <testmail@gmail.com
Subject: an example.com example email
Date: Mon, 7 Nov 2016 08:45:16

Dear Joe
Welcome to this example email. What a lovely day.

```

All settings are by default, the file is issued by the service
https://setup.mailu.io/2.0/

How can I make sure that only authorized users can send? Thank you!

**Comment by nextgens** [comment]
Please share your ``mailu.env`` (remove the SECRET_KEY) and ``docker-compose.yml`` file so that we can look into it.

**Reply by perec007** [reply]
[mailu.env.txt](https://github.com/Mailu/Mailu/files/13458613/mailu.env.txt)
[docker-compose.yml.txt](https://github.com/Mailu/Mailu/files/13458614/docker-compose.yml.txt)

**Reply by nextgens** [reply]
Hmm, you have some inconsistencies here: ``SCAN_MACROS=true`` but no section for oletools

That doesn't explain the "open-relay" thing though.

Are you running the curl command from the host or from elsewhere? It is expected that the host can send emails through Mailu without restrictions.

**Comment by perec007** [comment]
log in postfix container:


```
Nov 24 11:54:27 openmail postfix/smtpd[870]: connect from mailu-front-1.mailu_default[192.168.203.2]
Nov 24 11:54:27 openmail postfix/smtpd[870]: 0F0F42855C: client=unknown[192.168.203.1]
Nov 24 11:54:27 openmail postfix/cleanup[874]: 0F0F42855C: message-id=<>
Nov 24 11:54:27 openmail postfix/qmgr[376]: 0F0F42855C: from=<SRS0=1E2Q=HF=spam.spam=spam@myserver.com>, size=295, nrcpt=1 (queue active)
Nov 24 11:54:27 openmail postfix/lmtp[875]: 0F0F42855C: to=<infrait@myserver.com>, relay=front[192.168.203.2]:2525, delay=0.39, delays=0.27/0.04/0/0.08, dsn=2.0.0, status=sent (250 2.0.0 <infrait@myserver.com> +EcPFfOOYGVbAgAAKDix6Q Saved)
Nov 24 11:54:27 openmail postfix/qmgr[376]: 0F0F42855C: removed
Nov 24 11:54:27 openmail postfix/smtpd[870]: disconnect from unknown[192.168.203.1] ehlo=2 xclient=0/1 mail=1 rcpt=1 data=1 quit=1 commands=6/7
```

I am sending this message without authorization:

```
$ curl smtp://myserver.com --mail-from spam@spam.spam --mail-rcpt infrait@myserver.com -s --upload-file email.txt
$ cat email.txt
From: John Smith <spam@spam.biz>
To: Joe Smith <in@spam.com>
Subject: SPAM test
```

and see in main mailbox:
```
Delivered-To: <infrait@myserver.com>
Return-Path: <SRS0=1E2Q=HF=spam.spam=spam@myserver.com>
Received: from myserver.com ([192.168.203.7])
	by myserver.com with LMTP
	id MDiUK2SPYGVbAgAAKDix6Q:P1
	(envelope-from <SRS0=1E2Q=HF=spam.spam=spam@myserver.com>)
	for <infrait@myserver.com>; Fri, 24 Nov 2023 11:56:20 +0000
Received: from myserver.com ([192.168.203.7])
	by myserver.com with LMTP
	id MDiUK2SPYGVbAgAAKDix6Q
	(envelope-from <SRS0=1E2Q=HF=spam.spam=spam@myserver.com>)
	for <infrait@myserver.com>; Fri, 24 Nov 2023 11:56:20 +0000
Received: from email.txt (unknown [192.168.203.1])
	by myserver.com (Postfix) with ESMTP id 7710C2855C
	for <infrait@myserver.com>; Fri, 24 Nov 2023 11:56:20 +0000 (UTC)
Authentication-Results: myserver.com;
	none
From: John Smith <spam@spam.biz>
To: Joe Smith <in@spam.com>
Subject: SPAM test 
X-Spam-Level: ******
X-Spamd-Bar: ++++++
X-Spam: Yes

```

**Reply by nextgens** [reply]
``Received: from email.txt (unknown [192.168.203.1])``

... you are attempting this from the host. Try from a different IP address and you will see that it won't work

**Reply by perec007** [reply]
192.168.203.1  - this internal docker ip network, not real ip server or client

**Reply by nextgens** [reply]
So what is the IP of the client and why isn't Mailu seeing that?

**Reply by perec007** [reply]
Because Mailu is running in a docker container.
For mailu to see real addresses it must be running in net: host mode

**Reply by nextgens** [reply, marked as answer]
This does not make sense... Mailu should see the "right" source IP address.

Do you have the userland proxy enabled? If so, turn it off as per
https://github.com/Mailu/Mailu/discussions/3042#discussioncomment-7495362

**Reply by perec007** [reply]
great, thank you very much for the tip, this is what I need!
You need to add this to the installation instructions via docker and manuals :)
